exchanges/stream/stream_types.go

```diff
@@ -37,10 +39,11 @@ type Connection interface {
 	Shutdown() error
 }
 
-// Inspector is used to verify messages via SendMessageReturnResponse
-// Only one can used
+// Inspector is used to verify messages via SendMessageReturnResponsesWithInspection
 // It inspects the []bytes websocket message and returns true if it is the appropriate message to action
-type Inspector func([]byte) bool
+type Inspector interface {
+	Inspect([]byte) bool
```

Comment:
Soz; Nitpick:
`Inspect` doesn't feel appropriate for the boolean response type.
I'd say an Inspector would decide if things are Interesting, or Valid, or Appropriate.
So I guess I'm asking for an adjective.
`Appropriate` feels right

Reply:
For the use case I used `IsFinal` because it returns true if further processing is not required and updated the commentary [here](https://github.com/thrasher-corp/gocryptotrader/pull/1598/commits/6a2fbdec024a0e192a9255a6b23d9fd9eaf82338)

Reply:
That's good.